src/schema/posts.ts

```diff
@@ -2390,6 +2363,69 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         ),
       }));
     },
+    deleteSourcePostModeration: async (
+      _,
+      { postId }: { postId: string },
+      ctx: AuthContext,
+    ): Promise<GQLEmptyResponse> => {
+      const moderation = await ctx.con
```

Comment:
Here we can also add select statement

Reply:
also check source permission first before querying the moderation as done in edit?

Reply:
Yeah, makes sense. Let's do everything after that check.

Reply:
Ah, for this case, we can't since we need some values from the moderation object.